app/controllers/users/sessions_controller.rb

```diff
@@ -4,7 +4,20 @@ class Users::SessionsController < ApplicationController
 
   def new
     if (authentication = User::Authentication.find_by(token: params[:auth_token]))
-      return redirect_to sign_in_path if authentication.expired? || authentication.succeeded?
+      if authentication.expired?
+        authentication.reject reason: :expired
+        return redirect_to sign_in_path
+      end
+
+      if authentication.succeeded?
+        authentication.reject reason: :previously_succeeded
+        return redirect_to sign_in_path
+      end
+
+      unless authentication.user.admin? # Don't allow non-admins to sign in
+        authentication.reject reason: :non_admin
+        return redirect_to root_path
+      end
```

Comment:
This might be a bit confusing for someone who's not an admin and doesn't know what's going on.

Reply:
I also just remembered that https://github.com/hackclub/hackathons-backend/pull/77 added `Hackathons::Submissions#index` which allows signed-in non-admins to view their applications